azure-pipelines.yml

```diff
@@ -0,0 +1,217 @@
+#  - The DeployControl job is responsible for defining a variable that is used
+#  by the other deployment jobs to ensure that we only deploy for the OFFICIAL_REPO
+#  and not for PRs.
+#  - We currently use the template to run the tests (See azure-test-template.yml).
+#  - After the tests pass 3 Publish jobs go into action:
+#      - Documentation
+#        This job is executed on commits to master and tags.
+#        It deployes the new docs to the gh-pages branch at GitHub.
+#      - PyPI
+#        This job is executed on tags only.
+#        It deploy the package to the Python Package Index.
+#      - Anaconda
+#        This job is executed on commits to master and tags.
+#        It deploy the package to the DEV or TAG channels at Anaconda Cloud.
+############################################################################################
+
+# Build only for PRs for master branch
+pr:
+  autoCancel: true
+  branches:
+    include:
+      - master
+
+variables:
+  OFFICIAL_REPO: 'slaclab/pydm'
+
+jobs:
+  - template: azure-test-template.yml
+    parameters:
+      name: Linux
+      vmImage: 'Ubuntu 16.04'
+      build_docs: 1
+
+  - template: azure-test-template.yml
+    parameters:
+      name: Windows
+      vmImage: 'vs2017-win2016'
+
+  - template: azure-test-template.yml
+    parameters:
+      name: MacOS
+      vmImage: 'macOS-10.13'
+
+  - job: 'DeployControl'
+    pool:
+      vmImage: 'Ubuntu 16.04'
+    steps:
+    - bash: |
+        echo "##vso[task.setvariable variable=build_ok;isOutput=true]true"
+      name: var
+      condition: |
+        and
+        (
+          eq(variables['Build.Repository.Name'], variables['OFFICIAL_REPO']),
+          eq(variables['System.PullRequest.PullRequestNumber'], variables['NULL'])
+        )
+    - bash: echo $(var.build_ok)
+
+  - job: 'Publish_Documentation'
+    displayName: Publish - Documentation
+    condition: |
+      and
+      (
+        eq(dependencies.DeployControl.outputs['var.build_ok'], true),
+        or
+        (
+        contains(variables['Build.SourceBranch'],'heads/master'),
+        contains(variables['Build.SourceBranch'],'refs/tags')
+        )
+      )
+    dependsOn:
+      - DeployControl
+      - Linux
+      - MacOS
+      - Windows
+    pool:
+      vmImage: 'Ubuntu 16.04'
+
+    steps:
+    - checkout: self
+      persistCredentials: true
+      clean: true
+
+    - task: DownloadBuildArtifacts@0
+      inputs:
+        buildType: 'current'
+        artifactName: 'docs_html'
+        downloadPath: $(Build.Repository.LocalPath)/artifacts
+      displayName: Artifact Download - Documentation
+
+    - bash: |
+        git config --global user.name "Azure Pipelines"
+        git config --global user.email "azurebot@nonvalidemail.com"
+        git checkout gh-pages
+        cp -r $(Build.Repository.LocalPath)/artifacts/docs_html/* .
+        git add *
+        git commit -m "Updating Docs with latest build - ***NO_CI***"
```

Comment:
Nice!

Reply:
This one in my view was kind of ugly but it is what got me working so I kept it... suggestions are welcome! 🤣